lua/damagelogs/shared/events.lua

```diff
@@ -37,8 +37,9 @@ function Damagelog:AddEvent(event, f)
             end
         end
 
+        infos = util.Compress(util.TableToJSON(infos))
         net.Start("DL_RefreshDamagelog")
-        net.WriteTable(infos)
+        net.WriteData(infos, #infos)
```

Comment:
This should write the length of the data first, so that the client knows how much data to read.

The value of the `len` parameter in `net.Receive` is longer than the actual payload
![image](https://user-images.githubusercontent.com/5206198/105881239-290c4780-5ffc-11eb-992d-80d7f4378b7b.png)


Server side
```lua
net.WriteUInt(string.len(infos), 32)
net.WriteData(infos, string.len(infos))
```


Then on the client's end
```lua
local dataSize = net.ReadUInt(32)
local tbl = util.JSONToTable(util.Decompress(net.ReadData(dataSize)))
```

Reply:
done